### DISCUSSION TITLE
🐛 in props

### DISCUSSION THREAD

**Discussion by babur001** [vercel/next.js#27009] [Help] [answered]
### What version of Next.js are you using?

10.0.7

### What version of Node.js are you using?

14.15.4

### What browser are you using?

Chrome

### What operating system are you using?

Linux Ubuntu

### How are you deploying your application?

Vercel

### Describe the Bug

The bug is whenever I inspect the source code of [this site](view-source:https://www.instrument.uz/), I have noticed that in the bottom inside `<script id="__NEXT_DATA__" type="application/json">` I have all of the props passed from `getServerSideProps` _inside_ of this script 👀. 

Why is that? Is it just a feature or a bug? This is affecting the whole sites load timing

### Expected Behavior

_No Props_ json data, just generated HTML

### To Reproduce

No steps

**Comment by babur001** [comment]
The sites source: view-source:https://www.instrument.uz/

**Comment by timneutkens** [comment, marked as answer]
This is expected as the server-side react tree has to match with the client-side react tree for the process of `hydration` in React: https://reactjs.org/docs/react-dom.html#hydrate
